I just tried building with this on `Ubuntu 18.04` and `x86_64`:

```
==> Installing slate
==> slate: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'all' 'install' 'prefix=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4' 'mpi=1' 'blas=openblas' 'cuda=1' 'openmp=1' 'c_api=1' 'fortran_api=1' 'CXX=mpicxx' 'FC=mpif90'

3 errors found in build log:
     947    	-o unit_test/test_norm
     948    cd blaspp   && make install prefix=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4
     949    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/spack-src/blaspp'
     950    mkdir -p /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/include
     951    mkdir -p /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/lib
     952    cp include/*.{h,hh} /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/include
  >> 953    cp: cannot stat 'include/*.{h,hh}': No such file or directory
     954    GNUmakefile:115: recipe for target 'install' failed
  >> 955    make[1]: *** [install] Error 1
     956    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/spack-src/blaspp'
     957    GNUmakefile:644: recipe for target 'install' failed
  >> 958    make: *** [install] Error 2
     959    make: *** Waiting for unfinished jobs....

See build log for details:
  /tmp/root/spack-stage/spack-stage-slate-develop-jtpodvyve4waorvptkrkymb5xhomfod4/spack-build-out.txt
```

Full build out: [slate-build-out.txt](https://github.com/spack/spack/files/5295091/slate-build-out.txt)

--- a/var/spack/repos/builtin/packages/slate/package.py
+++ b/var/spack/repos/builtin/packages/slate/package.py
@@ -24,13 +24,9 @@ class Slate(Package):
     variant('cuda',   default=True, description='Build with CUDA support.')
     variant('mpi',    default=True, description='Build with MPI support.')
     variant('openmp', default=True, description='Build with OpenMP support.')
-    variant('blas',   default='openblas', description='Select BLAS implementation', 
-            values=('openblas', 'essl', 'mkl'), multi=False)
 
     depends_on('scalapack')
-    depends_on('intel-mkl', when='blas=mkl')
-    depends_on('essl',      when='blas=essl')
-    depends_on('openblas',  when='blas=openblas')
+    depends_on('blas')
     depends_on('cuda@9:10', when='+cuda')
     depends_on('mpi',       when='+mpi')
 
@@ -46,10 +42,20 @@ class Slate(Package):
             comp_cxx = 'mpicxx'
             comp_for = 'mpif90'
 
+        if '^openblas' in spec:
+            blas = 'openblas'
+        elif '^intel-mkl' in spec:
+            blas = 'mkl'
+        elif '^essl' in spec:
+            blas = 'essl'
+        else:
+            raise InstallError('Supports only BLAS provider openblas, intel-mkl, or essl')
+
+
         make('all', 'install',
              'prefix=' + prefix,
              'mpi=' + f_mpi,
-             'blas=' + spec.variants['blas'].value,
+             'blas=' + blas,
              'cuda=' + f_cuda,
              'openmp=' + f_omp,
              'c_api=1',

--- a/var/spack/repos/builtin/packages/slate/package.py
+++ b/var/spack/repos/builtin/packages/slate/package.py
@@ -49,8 +49,8 @@ class Slate(Package):
         elif '^essl' in spec:
             blas = 'essl'
         else:
-            raise InstallError('Supports only BLAS provider openblas, intel-mkl, or essl')
-
+            raise InstallError('Supports only BLAS provider '
+                               'openblas, intel-mkl, or essl')
 
         make('all', 'install',
              'prefix=' + prefix,

--- a/var/spack/repos/builtin/packages/slate/package.py
+++ b/var/spack/repos/builtin/packages/slate/package.py
@@ -27,6 +27,7 @@ class Slate(Package):
     variant('blas',   default='openblas', description='Select BLAS implementation', 
             values=('openblas', 'essl', 'mkl'), multi=False)
 
+    depends_on('bash', type='build')
     depends_on('scalapack')
     depends_on('intel-mkl', when='blas=mkl')
     depends_on('essl',      when='blas=essl')
@@ -47,6 +48,7 @@ class Slate(Package):
             comp_for = 'mpif90'
 
         make('all', 'install',
+             'SHELL=bash',
              'prefix=' + prefix,
              'mpi=' + f_mpi,
              'blas=' + spec.variants['blas'].value,

